src/modules/transformers/displace.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num_traits::Float;
+use math::{Point2, Point3, Point4};
+use NoiseModule;
+
+/// Noise Module that uses multiple source modules to displace each coordinate
+/// of the input value before returning the output value from the `source` module.
+pub struct Displace<Source, XDisplace, YDisplace, ZDisplace, UDisplace> {
```

Comment:
Do all dimensions all of these modules? I wonder if we could just have a Modules parameters, and then implement on tuples. Eg:

```
#[cfg_attr(rustfmt, rustfmt_skip)]
impl<Source, XDisplace, YDisplace, T> NoiseModule<Point2<T>>
    for Displace<Source, (XDisplace, YDisplace)>
    where Source: NoiseModule<Point2<T>, Output = T>,
          XDisplace: NoiseModule<Point2<T>, Output = T>,
          YDisplace: NoiseModule<Point2<T>, Output = T>,
          T: Float,
{
```